sdk/cosmos/azure_data_cosmos/src/clients/cosmos_client.rs

```diff
@@ -113,4 +142,25 @@ impl CosmosClientMethods for CosmosClient {
     fn database_client(&self, id: impl AsRef<str>) -> DatabaseClient {
         DatabaseClient::new(self.pipeline.clone(), &self.endpoint, id.as_ref())
     }
+
+    /// Gets the endpoint of the database account this client is connected to.
+    fn endpoint(&self) -> &Url {
+        &self.endpoint
+    }
+
+    fn query_databases(
+        &self,
+        query: impl Into<Query>,
+
+        #[allow(unused_variables)]
+        // REASON: This is a documented public API so prefixing with '_' is undesirable.
+        options: Option<QueryDatabasesOptions>,
```

Comment:
Can you have a different parameter name in the trait impl than in the trait itself?  Maybe like in C, where you put the parameter name in the .h file, but you omit it if it's unused in the .c file, you'd rename it to have the `_` here, but leave the original name in the Trait?

Reply:
Parameter names don't need to match between trait and impl, so this can absolutely just be `_options`, which suppresses the lint. The problem is that shows up that way in the docs, which I'd rather not do (I'll grab a screenshot and reply with it in a sec). I suppose it's not _harmful_ as all it does is hint to the user that the implementation doesn't actually **use** the parameter, but it feels... well... clunky. That's the intent of this "REASON" comment (which would eventually be the `reason` attribute of the `#[allow]` attribute, once we update our minimum rustc version to or beyond 1.81), to explain why we're doing an `#[allow]` here instead of prefixing with `_`.

Another option I've seen before, which I didn't like as much, is to "use" the parameter in a statement that will be compiled away. That looks something like this:

```rust
async fn create_item<T: Serialize + DeserializeOwned>(
    &self,
    partition_key: impl Into<PartitionKey>,
    item: T,
    options: Option<ItemOptions>,
) -> Result<Response<Item<T>>> {
    let _ = options;
    // .. the rest of the function
}
```

My personal thought was that I found that less clear than specifying exactly what lint we were addressing, but I'm open to it.

Reply:
Yeah, not necessarily suggesting we change it to that, just speculating on the reasoning behind it.